### DISCUSSION TITLE
A few questions

### DISCUSSION THREAD

**Discussion by Semisol** [owncast/owncast#1533] [Q&A]
I have a few questions that I'm wondering about:
- Can I IP ban people?
- Is there a way I can add moderators?
- Is there a way I can prevent other people from using my chat nickname?

If any of these are not currently doable but planned, I would like to know that.

**Comment by gabek** [comment]
> Can I IP ban people?

You cannot. Currently no IPs of users are stored.

> Is there a way I can add moderators?

This is being worked on for the next release.

> Is there a way I can prevent other people from using my chat nickname?

You could add your name to the banned username list in the chat settings.

**Reply by Semisol** [reply]
> You cannot. Currently no IPs of users are stored.

This would be good for ban evaders. Should I open an issue?

> You could add your name to the banned username list in the chat settings.

But it prevents me from changing my name to that, is there a way so that only I can use a nickname? (should I also open an issue?)

And is there planned support for badges for the streamer and mods? I think I saw it somewhere but not sure.

**Reply by gabek** [reply]
Yeah, absolutely open an issue for anything you'd like to see.  Though as it stands we purposefully don't store IP information to keep the amount of information we know about visitors to a minimum. I figured if IP-based banning bubbles up as something that's seen as required it can be discussed, but until then keeping people's IP addresses away from our data store is a good thing.

As for usernames, you could change your name and then add it to the list.

> And is there planned support for badges for the streamer and mods? I think I saw it somewhere but not sure.

There are no plans for this. Adding it for moderators would be simple, but I've never been a fan of elevating specific people in a conversation. Moderating is a responsibility, not a status.  As for the streamer, I can see that being a thing, but as it stands it's not something that's been discussed.

**Reply by Semisol** [reply]
> As for usernames, you could change your name and then add it to the list.

It would be annoying to do that every time, though.

> Yeah, absolutely open an issue for anything you'd like to see. Though as it stands we purposefully don't store IP information to keep the amount of information we know about visitors to a minimum.

I still think it should be added. At least hashes, or similar should be stored.

**Reply by gabek** [reply]
> It would be annoying to do that every time, though.

Just curious what the use case is here, why are you changing your name every time you join the chat?

**Reply by Semisol** [reply]
I'm not. When I reconnect by say reloading, the nickname will be banned, no?

**Reply by gabek** [reply]
Adding a username to the list will stop people from changing their name to anything in that list, it doesn't affect existing users or is a way to take action against anybody with any current names.  It's not a moderation tool, just a configuration.

**Reply by Semisol** [reply]
> Adding a username to the list will stop people from changing their name to anything in that list, it doesn't affect existing users or is a way to take action against anybody with any current names. It's not a moderation tool, just a configuration.

Okay, I think I get it.
I'll try it.

**Reply by Semisol** [reply]
@gabek I created an issue here: #1534
